modules/gcr__pull_repos/main.py

```diff
@@ -0,0 +1,198 @@
+                else:
+                    print('login failed')
+
+                registry_previous = registry_current
+            except Exception as e:
+                print(e, file=sys.stderr)
+
+        data['count'] = count
+        data['payload']['registry'] = docker_registry
+
+    except Exception as e:
+        print(e, file=sys.stderr)
+
+
+def gcr_pull(args, data):
+    try:
+        docker_client = docker.DockerClient(base_url=DOCKER_BASE_URL)
+        docker_registry = get_registry(args.get('repositories')[0])
+        docker_login_response = docker_login(docker_client, args.get('docker_username'), args.get('docker_password'), docker_registry)
+
+        if DOCKER_LOGIN_SUCCEEDED == docker_login_response.get('Status'):
+            count = 0
+            if len(args.get('repository_tags')) >= 1:
+                # pull provided tags
+                for tag in args.get('repository_tags'):
+                    repo = args.get('repositories')[0] + ':' + tag
+                    try:
+                        docker_pull(docker_client, repo)
+                        data['payload']['repository_tags'].append(tag)
+                        count += 1
+                    except Exception as e:
+                       print(e, file=sys.stderr)
+            else:
+                # pull all tags
+                try:
+                    repo = args.get('repositories')[0]
+                    docker_pull(docker_client, repo)
+                    count += 1
+                except Exception as e:
+                    print(e, file=sys.stderr)
+        else:
+            print('login failed')
+
+        data['count'] = count
+        data['payload']['repositories'] = args.get('repositories')
+
+    except Exception as e:
+        print(e, file=sys.stderr)
+
+
+def set_args(service_account_json_file_path=None, access_token=None, repositories=[], repository_tags=[]):
+    if service_account_json_file_path is not None:
+        # check the existence of service account json file
+        path = pathlib.Path(service_account_json_file_path)
+        if path.exists() is False:
+            print("Service account json file does not exist")
+            sys.exit(1)
+
+    args = {
+        'service_account_json_file_path': service_account_json_file_path,
+        'access_token': access_token,
+        'repositories': repositories,
+        'repository_tags': repository_tags
+    }
+
+    return args
+
+
+def main(args):
+    data  = {
+        'count': 0,
+        'payload': {
+            'repositories': [],
+            'repository_tags': []
+        }
+    }
+
+    docker_configure_username_password(args)
+
+    if len(args.get('repositories')) > 1:
+        gcr_pull_all(args, data)
+    else:
+        gcr_pull(args, data)
+
+    return data
+
+
+def summary(data):
+    out = ''
+    out += '{} GCR Repositories Pulled\n'.format(data['count']) 
+    out += 'GCR resources saved and use \'docker images\' command to check the result.\n'
```

Comment:
GCR resource saved. Use the \'docker images\'  command to check the results.

Reply:
Wonderfull, I will replace it.